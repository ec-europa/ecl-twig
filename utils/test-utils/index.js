const Twig = require('twig');
const { format } = require('prettier');

const renderTwigFile = (file, options, cb) =>
  Twig.renderFile(file, options, (err, html) =>
    cb(err, format(html, { parser: 'html' }))
  );

const renderTwigFileAsNode = (file, options) =>
  new Promise((resolve, reject) =>
    Twig.renderFile(file, options, (err, html) => {
      if (err) return reject(err);

      const div = document.createElement('div');
      div.innerHTML = html.trim();

      return resolve(div.firstChild);
    })
  );

module.exports = {
  renderTwigFile,
  renderTwigFileAsNode,
};
