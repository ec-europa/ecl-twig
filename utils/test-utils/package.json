{
  "name": "@ecl-twig/test-utils",
  "version": "2.29.0",
  "description": "A local test helper",
  "main": "index.js",
  "private": true,
  "devDependencies": {
    "deepmerge": "4.2.2"
  }
}
