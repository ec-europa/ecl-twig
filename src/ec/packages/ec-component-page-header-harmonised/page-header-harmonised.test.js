import { merge, renderTwigFileAsNode } from '@ecl-twig/test-utils';

import breadcrumbDataSimple from '@ecl/ec-specs-breadcrumb-core/demo/data-simple';

import pageHeaderHarmonisedDataTitle from '@ecl/ec-specs-page-header-harmonised/demo/data--title';
import pageHeaderHarmonisedDataMetaTitle from '@ecl/ec-specs-page-header-harmonised/demo/data--meta-title';
import pageHeaderHarmonisedDataMetaTitleDescription from '@ecl/ec-specs-page-header-harmonised/demo/data--meta-title-description';

function formatBreadcrumbLink(l) {
  const link = {
    label: l.label,
    path: l.href,
  };

  return link;
}

function formatPageHeaderHarmonisedInfo(i) {
  const iconType = i.icon.split('--');
  const info = {
    text: i.text,
    icon: {
      type: iconType[0],
      name: iconType[1],
      path: 'static/icons.svg',
    },
  };

  return info;
}

function preparePageHeaderHarmonisedData(data) {
  const output = {};

  output.breadcrumb = {
    links: breadcrumbDataSimple.items.map(formatBreadcrumbLink),
    navigation_text: breadcrumbDataSimple.label,
    icon_file_path: 'static/icons.svg',
  };

  output.title = data.title;

  if (data.description) {
    output.description = data.description;
  }

  if (data.meta) {
    output.meta = data.meta;
  }

  if (data.infos) {
    output.infos = data.infos.map(formatPageHeaderHarmonisedInfo);
  }

  return output;
}

describe('EC - Page Header Harmonised', () => {
  const template =
    '@ecl-twig/ec-component-page-header-harmonised/page-header-harmonised.html.twig';
  const render = params => renderTwigFileAsNode(template, params);

  describe('title', () => {
    const data = preparePageHeaderHarmonisedData(pageHeaderHarmonisedDataTitle);

    test(`- renders correctly`, () => {
      expect.assertions(1);
      return expect(render(data)).resolves.toMatchSnapshot();
    });

    test('renders correctly with extra class names', () => {
      expect.assertions(1);

      const withExtraClasses = merge(data, {
        extra_classes: 'custom-class custom-class--test',
      });

      return expect(render(withExtraClasses)).resolves.toMatchSnapshot();
    });

    test('renders correctly with extra attributes', () => {
      expect.assertions(1);

      const withExtraAttributes = merge(data, {
        extra_attributes: [
          { name: 'data-test', value: 'data-test-value' },
          { name: 'data-test-1', value: 'data-test-value-1' },
        ],
      });

      return expect(render(withExtraAttributes)).resolves.toMatchSnapshot();
    });
  });

  describe('meta-title', () => {
    const data = preparePageHeaderHarmonisedData(
      pageHeaderHarmonisedDataMetaTitle
    );

    test(`- renders correctly`, () => {
      expect.assertions(1);
      return expect(render(data)).resolves.toMatchSnapshot();
    });

    test('renders correctly with extra class names', () => {
      expect.assertions(1);

      const withExtraClasses = merge(data, {
        extra_classes: 'custom-class custom-class--test',
      });

      return expect(render(withExtraClasses)).resolves.toMatchSnapshot();
    });

    test('renders correctly with extra attributes', () => {
      expect.assertions(1);

      const withExtraAttributes = merge(data, {
        extra_attributes: [
          { name: 'data-test', value: 'data-test-value' },
          { name: 'data-test-1', value: 'data-test-value-1' },
        ],
      });

      return expect(render(withExtraAttributes)).resolves.toMatchSnapshot();
    });
  });

  describe('meta-title-description', () => {
    const data = preparePageHeaderHarmonisedData(
      pageHeaderHarmonisedDataMetaTitleDescription
    );

    test(`- renders correctly`, () => {
      expect.assertions(1);
      return expect(render(data)).resolves.toMatchSnapshot();
    });

    test('renders correctly with extra class names', () => {
      expect.assertions(1);

      const withExtraClasses = merge(data, {
        extra_classes: 'custom-class custom-class--test',
      });

      return expect(render(withExtraClasses)).resolves.toMatchSnapshot();
    });

    test('renders correctly with extra attributes', () => {
      expect.assertions(1);

      const withExtraAttributes = merge(data, {
        extra_attributes: [
          { name: 'data-test', value: 'data-test-value' },
          { name: 'data-test-1', value: 'data-test-value-1' },
        ],
      });

      return expect(render(withExtraAttributes)).resolves.toMatchSnapshot();
    });
  });
});
