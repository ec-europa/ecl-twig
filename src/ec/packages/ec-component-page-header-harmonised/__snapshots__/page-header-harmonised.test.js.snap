// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EC - Page Header Harmonised meta-title - renders correctly 1`] = `
<div
  class="ecl-page-header-harmonised"
>
  <div
    class="ecl-container"
  >
    <nav
      aria-label="You are here:"
      class="ecl-breadcrumb ecl-page-header-harmonised__breadcrumb"
      data-ecl-auto-init="Breadcrumb"
      data-ecl-breadcrumb="true"
    >
      <ol
        class="ecl-breadcrumb__container"
      >
        <li
          aria-hidden="false"
          class="ecl-breadcrumb__segment"
          data-ecl-breadcrumb-item="static"
        >
          <a
            class="ecl-link ecl-link--standalone ecl-breadcrumb__link"
            href="/example"
          >
            Home
          </a>
          <svg
            aria-hidden="true"
            class="ecl-icon ecl-icon--xs ecl-icon--rotate-90 ecl-icon--inverted ecl-breadcrumb__icon"
            focusable="false"
            role="presentation"
          >
            <use
              xlink:href="static/icons.svg#ui--corner-arrow"
            />
          </svg>
        </li>
        <li
          aria-hidden="false"
          class="ecl-breadcrumb__segment"
          data-ecl-breadcrumb-item="static"
        >
          <a
            class="ecl-link ecl-link--standalone ecl-breadcrumb__link"
            href="/example"
          >
            About the European Commission
          </a>
          <svg
            aria-hidden="true"
            class="ecl-icon ecl-icon--xs ecl-icon--rotate-90 ecl-icon--inverted ecl-breadcrumb__icon"
            focusable="false"
            role="presentation"
          >
            <use
              xlink:href="static/icons.svg#ui--corner-arrow"
            />
          </svg>
        </li>
        <li
          aria-current="page"
          aria-hidden="false"
          class="ecl-breadcrumb__segment ecl-breadcrumb__current-page"
          data-ecl-breadcrumb-item="static"
        >
          News
        </li>
      </ol>
    </nav>
    <div
      class="ecl-page-header-harmonised__meta"
    >
      Meta info
    </div>
    <h1
      class="ecl-page-header-harmonised__title"
    >
      Page title
    </h1>
  </div>
</div>
`;

exports[`EC - Page Header Harmonised meta-title renders correctly with extra attributes 1`] = `
<div
  class="ecl-page-header-harmonised"
  data-test="data-test-value"
  data-test-1="data-test-value-1"
>
  <div
    class="ecl-container"
  >
    <nav
      aria-label="You are here:"
      class="ecl-breadcrumb ecl-page-header-harmonised__breadcrumb"
      data-ecl-auto-init="Breadcrumb"
      data-ecl-breadcrumb="true"
    >
      <ol
        class="ecl-breadcrumb__container"
      >
        <li
          aria-hidden="false"
          class="ecl-breadcrumb__segment"
          data-ecl-breadcrumb-item="static"
        >
          <a
            class="ecl-link ecl-link--standalone ecl-breadcrumb__link"
            href="/example"
          >
            Home
          </a>
          <svg
            aria-hidden="true"
            class="ecl-icon ecl-icon--xs ecl-icon--rotate-90 ecl-icon--inverted ecl-breadcrumb__icon"
            focusable="false"
            role="presentation"
          >
            <use
              xlink:href="static/icons.svg#ui--corner-arrow"
            />
          </svg>
        </li>
        <li
          aria-hidden="false"
          class="ecl-breadcrumb__segment"
          data-ecl-breadcrumb-item="static"
        >
          <a
            class="ecl-link ecl-link--standalone ecl-breadcrumb__link"
            href="/example"
          >
            About the European Commission
          </a>
          <svg
            aria-hidden="true"
            class="ecl-icon ecl-icon--xs ecl-icon--rotate-90 ecl-icon--inverted ecl-breadcrumb__icon"
            focusable="false"
            role="presentation"
          >
            <use
              xlink:href="static/icons.svg#ui--corner-arrow"
            />
          </svg>
        </li>
        <li
          aria-current="page"
          aria-hidden="false"
          class="ecl-breadcrumb__segment ecl-breadcrumb__current-page"
          data-ecl-breadcrumb-item="static"
        >
          News
        </li>
      </ol>
    </nav>
    <div
      class="ecl-page-header-harmonised__meta"
    >
      Meta info
    </div>
    <h1
      class="ecl-page-header-harmonised__title"
    >
      Page title
    </h1>
  </div>
</div>
`;

exports[`EC - Page Header Harmonised meta-title renders correctly with extra class names 1`] = `
<div
  class="ecl-page-header-harmonised custom-class custom-class--test"
>
  <div
    class="ecl-container"
  >
    <nav
      aria-label="You are here:"
      class="ecl-breadcrumb ecl-page-header-harmonised__breadcrumb"
      data-ecl-auto-init="Breadcrumb"
      data-ecl-breadcrumb="true"
    >
      <ol
        class="ecl-breadcrumb__container"
      >
        <li
          aria-hidden="false"
          class="ecl-breadcrumb__segment"
          data-ecl-breadcrumb-item="static"
        >
          <a
            class="ecl-link ecl-link--standalone ecl-breadcrumb__link"
            href="/example"
          >
            Home
          </a>
          <svg
            aria-hidden="true"
            class="ecl-icon ecl-icon--xs ecl-icon--rotate-90 ecl-icon--inverted ecl-breadcrumb__icon"
            focusable="false"
            role="presentation"
          >
            <use
              xlink:href="static/icons.svg#ui--corner-arrow"
            />
          </svg>
        </li>
        <li
          aria-hidden="false"
          class="ecl-breadcrumb__segment"
          data-ecl-breadcrumb-item="static"
        >
          <a
            class="ecl-link ecl-link--standalone ecl-breadcrumb__link"
            href="/example"
          >
            About the European Commission
          </a>
          <svg
            aria-hidden="true"
            class="ecl-icon ecl-icon--xs ecl-icon--rotate-90 ecl-icon--inverted ecl-breadcrumb__icon"
            focusable="false"
            role="presentation"
          >
            <use
              xlink:href="static/icons.svg#ui--corner-arrow"
            />
          </svg>
        </li>
        <li
          aria-current="page"
          aria-hidden="false"
          class="ecl-breadcrumb__segment ecl-breadcrumb__current-page"
          data-ecl-breadcrumb-item="static"
        >
          News
        </li>
      </ol>
    </nav>
    <div
      class="ecl-page-header-harmonised__meta"
    >
      Meta info
    </div>
    <h1
      class="ecl-page-header-harmonised__title"
    >
      Page title
    </h1>
  </div>
</div>
`;

exports[`EC - Page Header Harmonised meta-title-description - renders correctly 1`] = `
<div
  class="ecl-page-header-harmonised"
>
  <div
    class="ecl-container"
  >
    <nav
      aria-label="You are here:"
      class="ecl-breadcrumb ecl-page-header-harmonised__breadcrumb"
      data-ecl-auto-init="Breadcrumb"
      data-ecl-breadcrumb="true"
    >
      <ol
        class="ecl-breadcrumb__container"
      >
        <li
          aria-hidden="false"
          class="ecl-breadcrumb__segment"
          data-ecl-breadcrumb-item="static"
        >
          <a
            class="ecl-link ecl-link--standalone ecl-breadcrumb__link"
            href="/example"
          >
            Home
          </a>
          <svg
            aria-hidden="true"
            class="ecl-icon ecl-icon--xs ecl-icon--rotate-90 ecl-icon--inverted ecl-breadcrumb__icon"
            focusable="false"
            role="presentation"
          >
            <use
              xlink:href="static/icons.svg#ui--corner-arrow"
            />
          </svg>
        </li>
        <li
          aria-hidden="false"
          class="ecl-breadcrumb__segment"
          data-ecl-breadcrumb-item="static"
        >
          <a
            class="ecl-link ecl-link--standalone ecl-breadcrumb__link"
            href="/example"
          >
            About the European Commission
          </a>
          <svg
            aria-hidden="true"
            class="ecl-icon ecl-icon--xs ecl-icon--rotate-90 ecl-icon--inverted ecl-breadcrumb__icon"
            focusable="false"
            role="presentation"
          >
            <use
              xlink:href="static/icons.svg#ui--corner-arrow"
            />
          </svg>
        </li>
        <li
          aria-current="page"
          aria-hidden="false"
          class="ecl-breadcrumb__segment ecl-breadcrumb__current-page"
          data-ecl-breadcrumb-item="static"
        >
          News
        </li>
      </ol>
    </nav>
    <div
      class="ecl-page-header-harmonised__meta"
    >
      Meta info
    </div>
    <h1
      class="ecl-page-header-harmonised__title"
    >
      Page title
    </h1>
    <p
      class="ecl-page-header-harmonised__description"
    >
      Sed ut perspiciatis unde omnis iste natus error sit voluptatem accusantium doloremque laudantium
    </p>
  </div>
</div>
`;

exports[`EC - Page Header Harmonised meta-title-description renders correctly with extra attributes 1`] = `
<div
  class="ecl-page-header-harmonised"
  data-test="data-test-value"
  data-test-1="data-test-value-1"
>
  <div
    class="ecl-container"
  >
    <nav
      aria-label="You are here:"
      class="ecl-breadcrumb ecl-page-header-harmonised__breadcrumb"
      data-ecl-auto-init="Breadcrumb"
      data-ecl-breadcrumb="true"
    >
      <ol
        class="ecl-breadcrumb__container"
      >
        <li
          aria-hidden="false"
          class="ecl-breadcrumb__segment"
          data-ecl-breadcrumb-item="static"
        >
          <a
            class="ecl-link ecl-link--standalone ecl-breadcrumb__link"
            href="/example"
          >
            Home
          </a>
          <svg
            aria-hidden="true"
            class="ecl-icon ecl-icon--xs ecl-icon--rotate-90 ecl-icon--inverted ecl-breadcrumb__icon"
            focusable="false"
            role="presentation"
          >
            <use
              xlink:href="static/icons.svg#ui--corner-arrow"
            />
          </svg>
        </li>
        <li
          aria-hidden="false"
          class="ecl-breadcrumb__segment"
          data-ecl-breadcrumb-item="static"
        >
          <a
            class="ecl-link ecl-link--standalone ecl-breadcrumb__link"
            href="/example"
          >
            About the European Commission
          </a>
          <svg
            aria-hidden="true"
            class="ecl-icon ecl-icon--xs ecl-icon--rotate-90 ecl-icon--inverted ecl-breadcrumb__icon"
            focusable="false"
            role="presentation"
          >
            <use
              xlink:href="static/icons.svg#ui--corner-arrow"
            />
          </svg>
        </li>
        <li
          aria-current="page"
          aria-hidden="false"
          class="ecl-breadcrumb__segment ecl-breadcrumb__current-page"
          data-ecl-breadcrumb-item="static"
        >
          News
        </li>
      </ol>
    </nav>
    <div
      class="ecl-page-header-harmonised__meta"
    >
      Meta info
    </div>
    <h1
      class="ecl-page-header-harmonised__title"
    >
      Page title
    </h1>
    <p
      class="ecl-page-header-harmonised__description"
    >
      Sed ut perspiciatis unde omnis iste natus error sit voluptatem accusantium doloremque laudantium
    </p>
  </div>
</div>
`;

exports[`EC - Page Header Harmonised meta-title-description renders correctly with extra class names 1`] = `
<div
  class="ecl-page-header-harmonised custom-class custom-class--test"
>
  <div
    class="ecl-container"
  >
    <nav
      aria-label="You are here:"
      class="ecl-breadcrumb ecl-page-header-harmonised__breadcrumb"
      data-ecl-auto-init="Breadcrumb"
      data-ecl-breadcrumb="true"
    >
      <ol
        class="ecl-breadcrumb__container"
      >
        <li
          aria-hidden="false"
          class="ecl-breadcrumb__segment"
          data-ecl-breadcrumb-item="static"
        >
          <a
            class="ecl-link ecl-link--standalone ecl-breadcrumb__link"
            href="/example"
          >
            Home
          </a>
          <svg
            aria-hidden="true"
            class="ecl-icon ecl-icon--xs ecl-icon--rotate-90 ecl-icon--inverted ecl-breadcrumb__icon"
            focusable="false"
            role="presentation"
          >
            <use
              xlink:href="static/icons.svg#ui--corner-arrow"
            />
          </svg>
        </li>
        <li
          aria-hidden="false"
          class="ecl-breadcrumb__segment"
          data-ecl-breadcrumb-item="static"
        >
          <a
            class="ecl-link ecl-link--standalone ecl-breadcrumb__link"
            href="/example"
          >
            About the European Commission
          </a>
          <svg
            aria-hidden="true"
            class="ecl-icon ecl-icon--xs ecl-icon--rotate-90 ecl-icon--inverted ecl-breadcrumb__icon"
            focusable="false"
            role="presentation"
          >
            <use
              xlink:href="static/icons.svg#ui--corner-arrow"
            />
          </svg>
        </li>
        <li
          aria-current="page"
          aria-hidden="false"
          class="ecl-breadcrumb__segment ecl-breadcrumb__current-page"
          data-ecl-breadcrumb-item="static"
        >
          News
        </li>
      </ol>
    </nav>
    <div
      class="ecl-page-header-harmonised__meta"
    >
      Meta info
    </div>
    <h1
      class="ecl-page-header-harmonised__title"
    >
      Page title
    </h1>
    <p
      class="ecl-page-header-harmonised__description"
    >
      Sed ut perspiciatis unde omnis iste natus error sit voluptatem accusantium doloremque laudantium
    </p>
  </div>
</div>
`;

exports[`EC - Page Header Harmonised title - renders correctly 1`] = `
<div
  class="ecl-page-header-harmonised"
>
  <div
    class="ecl-container"
  >
    <nav
      aria-label="You are here:"
      class="ecl-breadcrumb ecl-page-header-harmonised__breadcrumb"
      data-ecl-auto-init="Breadcrumb"
      data-ecl-breadcrumb="true"
    >
      <ol
        class="ecl-breadcrumb__container"
      >
        <li
          aria-hidden="false"
          class="ecl-breadcrumb__segment"
          data-ecl-breadcrumb-item="static"
        >
          <a
            class="ecl-link ecl-link--standalone ecl-breadcrumb__link"
            href="/example"
          >
            Home
          </a>
          <svg
            aria-hidden="true"
            class="ecl-icon ecl-icon--xs ecl-icon--rotate-90 ecl-icon--inverted ecl-breadcrumb__icon"
            focusable="false"
            role="presentation"
          >
            <use
              xlink:href="static/icons.svg#ui--corner-arrow"
            />
          </svg>
        </li>
        <li
          aria-hidden="false"
          class="ecl-breadcrumb__segment"
          data-ecl-breadcrumb-item="static"
        >
          <a
            class="ecl-link ecl-link--standalone ecl-breadcrumb__link"
            href="/example"
          >
            About the European Commission
          </a>
          <svg
            aria-hidden="true"
            class="ecl-icon ecl-icon--xs ecl-icon--rotate-90 ecl-icon--inverted ecl-breadcrumb__icon"
            focusable="false"
            role="presentation"
          >
            <use
              xlink:href="static/icons.svg#ui--corner-arrow"
            />
          </svg>
        </li>
        <li
          aria-current="page"
          aria-hidden="false"
          class="ecl-breadcrumb__segment ecl-breadcrumb__current-page"
          data-ecl-breadcrumb-item="static"
        >
          News
        </li>
      </ol>
    </nav>
    <h1
      class="ecl-page-header-harmonised__title"
    >
      Page title
    </h1>
  </div>
</div>
`;

exports[`EC - Page Header Harmonised title renders correctly with extra attributes 1`] = `
<div
  class="ecl-page-header-harmonised"
  data-test="data-test-value"
  data-test-1="data-test-value-1"
>
  <div
    class="ecl-container"
  >
    <nav
      aria-label="You are here:"
      class="ecl-breadcrumb ecl-page-header-harmonised__breadcrumb"
      data-ecl-auto-init="Breadcrumb"
      data-ecl-breadcrumb="true"
    >
      <ol
        class="ecl-breadcrumb__container"
      >
        <li
          aria-hidden="false"
          class="ecl-breadcrumb__segment"
          data-ecl-breadcrumb-item="static"
        >
          <a
            class="ecl-link ecl-link--standalone ecl-breadcrumb__link"
            href="/example"
          >
            Home
          </a>
          <svg
            aria-hidden="true"
            class="ecl-icon ecl-icon--xs ecl-icon--rotate-90 ecl-icon--inverted ecl-breadcrumb__icon"
            focusable="false"
            role="presentation"
          >
            <use
              xlink:href="static/icons.svg#ui--corner-arrow"
            />
          </svg>
        </li>
        <li
          aria-hidden="false"
          class="ecl-breadcrumb__segment"
          data-ecl-breadcrumb-item="static"
        >
          <a
            class="ecl-link ecl-link--standalone ecl-breadcrumb__link"
            href="/example"
          >
            About the European Commission
          </a>
          <svg
            aria-hidden="true"
            class="ecl-icon ecl-icon--xs ecl-icon--rotate-90 ecl-icon--inverted ecl-breadcrumb__icon"
            focusable="false"
            role="presentation"
          >
            <use
              xlink:href="static/icons.svg#ui--corner-arrow"
            />
          </svg>
        </li>
        <li
          aria-current="page"
          aria-hidden="false"
          class="ecl-breadcrumb__segment ecl-breadcrumb__current-page"
          data-ecl-breadcrumb-item="static"
        >
          News
        </li>
      </ol>
    </nav>
    <h1
      class="ecl-page-header-harmonised__title"
    >
      Page title
    </h1>
  </div>
</div>
`;

exports[`EC - Page Header Harmonised title renders correctly with extra class names 1`] = `
<div
  class="ecl-page-header-harmonised custom-class custom-class--test"
>
  <div
    class="ecl-container"
  >
    <nav
      aria-label="You are here:"
      class="ecl-breadcrumb ecl-page-header-harmonised__breadcrumb"
      data-ecl-auto-init="Breadcrumb"
      data-ecl-breadcrumb="true"
    >
      <ol
        class="ecl-breadcrumb__container"
      >
        <li
          aria-hidden="false"
          class="ecl-breadcrumb__segment"
          data-ecl-breadcrumb-item="static"
        >
          <a
            class="ecl-link ecl-link--standalone ecl-breadcrumb__link"
            href="/example"
          >
            Home
          </a>
          <svg
            aria-hidden="true"
            class="ecl-icon ecl-icon--xs ecl-icon--rotate-90 ecl-icon--inverted ecl-breadcrumb__icon"
            focusable="false"
            role="presentation"
          >
            <use
              xlink:href="static/icons.svg#ui--corner-arrow"
            />
          </svg>
        </li>
        <li
          aria-hidden="false"
          class="ecl-breadcrumb__segment"
          data-ecl-breadcrumb-item="static"
        >
          <a
            class="ecl-link ecl-link--standalone ecl-breadcrumb__link"
            href="/example"
          >
            About the European Commission
          </a>
          <svg
            aria-hidden="true"
            class="ecl-icon ecl-icon--xs ecl-icon--rotate-90 ecl-icon--inverted ecl-breadcrumb__icon"
            focusable="false"
            role="presentation"
          >
            <use
              xlink:href="static/icons.svg#ui--corner-arrow"
            />
          </svg>
        </li>
        <li
          aria-current="page"
          aria-hidden="false"
          class="ecl-breadcrumb__segment ecl-breadcrumb__current-page"
          data-ecl-breadcrumb-item="static"
        >
          News
        </li>
      </ol>
    </nav>
    <h1
      class="ecl-page-header-harmonised__title"
    >
      Page title
    </h1>
  </div>
</div>
`;
